src/framework/preferences/js/Builder.js

```diff
@@ -196,12 +196,22 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
     };
 
     /*
-     * A one-stop-shop function to build and instantiate a prefsEditor from a schema.
+     * A one-stop-shop component to build and instantiate a prefsEditor from a schema.
      */
-    fluid.prefs.create = function (container, options) {
-        options = options || {};
-        var builder = fluid.prefs.builder(options.build);
-        return fluid.invokeGlobalFunction(builder.options.assembledPrefsEditorGrade, [container, options.prefsEditor]);
-    };
+    fluid.defaults("fluid.prefs.create", {
+        gradeNames: ["fluid.viewComponent"],
+        builder: {}, // options provided by users
+        prefsEditor: {}, // options provided by users
+        members: {
+            constructedGrades: "@expand:fluid.prefs.builder({that}.options.build)"
+        },
+        components: {
+            prefsEditor: {
+                type: "{fluid.prefs.create}.constructedGrades.options.assembledPrefsEditorGrade",
```

Comment:
@cindyli this is an interesting approach. I wonder if you could do this with an invoker and dynamic grade instead? But i'm also not sure that that would be a better solution than what's here. @amb26 do you have any thoughts on this approach.